src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,57 +1,145 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { memo, useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
-  return (
-    <>
-      <form className="find-movie">
-        <div className="field">
-          <label className="label" htmlFor="movie-title">
-            Movie title
-          </label>
-
-          <div className="control">
-            <input
-              data-cy="titleField"
-              type="text"
-              id="movie-title"
-              placeholder="Enter a title to search"
-              className="input is-dander"
-            />
+type Props = {
+  onAddMovie: React.Dispatch<React.SetStateAction<Movie[]>>;
+};
+
+const defaultPicture = (
+  'https://via.placeholder.com/360x270.png?text=no%20preview'
+);
+
+export const FindMovie: React.FC<Props> = memo(
+  ({ onAddMovie }) => {
+    const [movie, setMovie] = useState<Movie | null>(null);
+    const [inputValue, setInputValue] = useState('');
+    const [isError, setIsError] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const handleClickSumbit = () => {
+      if (inputValue) {
+        setIsLoading(true);
+
+        getMovie(inputValue)
+          .then(movie_API => {
+            if ('Error' in movie_API) {
+              setIsError(true);
+            } else {
+              setMovie({
+                title: movie_API.Title,
+                description: movie_API.Plot,
+                imgUrl: movie_API.Poster !== 'N/A'
+                  ? movie_API.Poster
+                  : defaultPicture,
+                imdbUrl: `https://www.imdb.com/title/${movie_API.imdbID}`,
+                imdbId: movie_API.imdbID,
+              });
+            }
+          })
+          .finally(() => setIsLoading(false));
+      }
+    };
+
+    const clear = () => {
+      setMovie(null);
+      setInputValue('');
+    };
+
+    const handleClickAddMovie = () => {
+      onAddMovie((prev) => {
+        if (movie && !prev.some(m => m.imdbId === movie.imdbId)) {
+          return [...prev, movie];
+        }
+
+        return prev;
+      });
+
+      clear();
+    };
```

Comment:
let's move this method to parent component and pass it here as props

and do not forget to display an error when the film is already in the list